src/layer/place.js

```diff
@@ -187,7 +181,31 @@ export const city = {
         [11, 0.9],
       ],
     },
-    "text-field": Label.localizedName,
+    "text-field": [
+      "let",
+      "localizedName",
+      "",
+      "localizedCollator",
+      ["collator", {}],
+      [
+        "case",
+        [
+          "==",
+          ["var", "localizedName"],
+          ["get", "name"],
+          ["var", "localizedCollator"],
+        ],
```

Comment:
Sometimes there are multiple local names, but one of them matches the name in the user-preferred language, resulting in a partly redundant gloss:

[<img src="https://user-images.githubusercontent.com/1231218/204645936-748ee257-e61b-4a8d-b17e-5be84547432c.png" width="400" alt="Casablanca">](https://zelonewolf.github.io/openstreetmap-americana/#map=6.85/33.631/-7.526&language=en) [<img src="https://user-images.githubusercontent.com/1231218/204646103-926ae62b-e06b-44d8-b804-4d8325c4a533.png" width="400" alt="Hong Kong">](https://zelonewolf.github.io/openstreetmap-americana/#map=9.5/22.2828/114.1545&language=en)

Multiple local names occur in multilingual cities, but sometimes the same thing happens for other reasons too:

[<img src="https://user-images.githubusercontent.com/1231218/204645570-eb5fd10e-6fd8-44a4-b021-8eed5835d9c5.png" width="Derry/Londonderry">](https://zelonewolf.github.io/openstreetmap-americana/#map=8/54.985/-7.407&language=en)

I’m not sure we can solve this problem on the client side. The style specification lacks an operator for splitting a string (mapbox/mapbox-gl-js#8526) or testing a string against a regular expression (mapbox/mapbox-gl-js#4089). It isn’t enough to just search for the user-preferred name inside the local name, because of cases where the local name adds a prefix or suffix. Cleaning up the abridged local name would be messy anyways.

Ideally, when a place has multiple names, `name` would be a [semicolon-delimited list](https://wiki.openstreetmap.org/wiki/Semi-colon_value_separator), which would make it much easier for renderers to pretty-print the values using a delimiter appropriate to the style. (Some existing renderers would automatically replace the semicolon with an em dash, which would look great.) Unfortunately, most multilingual local communities have [standardized](https://wiki.openstreetmap.org/wiki/Multilingual_names) on formats delimited by a hyphen-minus (`-`) or solidus (`/`), optionally surrounded by spaces, or just a space with no delimiter.

Reply:
I'm willing to say that the practice of encoding multiple languages in a name label is just a poor practice and the fact that it looks ugly can't be helped.

Reply:
If we can do something reasonable with semicolon delimited names, that would be good.  I'm in favor of encouraging that practice over other non-standard delimiters.  Although I think multiple names in the name field should be avoided as much as possible.

Reply:
> I'm willing to say that the practice of encoding multiple languages in a name label is just a poor practice and the fact that it looks ugly can't be helped.

I think it’s a fine practice to put multiple values in `name` as long as there’s a good reason. Good reasons include local multilingualism (e.g., Hong Kong), [multigraphia](https://en.wikipedia.org/wiki/Digraphia) (Casablanca), or intractable political disputes ([Derry/Londonderry](https://en.wikipedia.org/wiki/Derry/Londonderry_name_dispute), the subject of [one of Wikipedia’s lamest edit wars](https://en.wikipedia.org/wiki/Wikipedia:Lamest_edit_wars#Derry/Londonderry)). Rand McNally is able to pick a side in https://github.com/ZeLonewolf/openstreetmap-americana/pull/592#discussion_r1035511841 because they curate place names by hand. By contrast, we’re developing blanket rules to run automatically, and we don’t have many tools for text processing at this level of the stack.

That said, the use of delimiters other than a semicolon is a problem. The dashes and other devices are a historical anomaly largely predating the widespread acceptance of the semicolon as a value delimiter; keys like [`name_1`](https://wiki.openstreetmap.org/wiki/Proposed_features/Remove_suffixed_name-tags_from_wiki) were still common back then. At this point, I think the continuing use of dashes and slashes as tagging for the renderer – namely, for openstreetmap-carto, at the expense of other data consumers [like Nominatim](https://nominatim.org/2021/10/10/detecting-languages.html). But even openstreetmap-carto’s developers agree that a feature should at least contain enough information to parse the multivalued `name`: https://github.com/gravitystorm/openstreetmap-carto/issues/4074#issuecomment-773980684.

Nowadays, [`default_language`](https://wiki.openstreetmap.org/wiki/Key:default_language) can be a format string, but I’m not optimistic about the prospects of parsing `name` based on a `default_language` format string using the existing expression operators. If a local community chooses to use a semicolon as its delimiter, then `default_language` would be both a format string and a humble multivalued list. It would become much more feasible for expressions in this style to find and omit the redundant name without the risk of truncating a name that contains a dash. We could easily replace a finite number of remaining semicolons with em dashes.

I’d be inclined to merge the PR with this problem remaining, if only as a showcase for why local communities should consider a more structured representation of `name` when they need to avoid favoring one local name over another.